### DISCUSSION TITLE
How are custom modules created in the Synthea Module Builder evaluated?

### DISCUSSION THREAD

**Discussion by tamasVarjas** [synthetichealth/synthea#1040] [Q&A] [answered]
This might be a beginner question and it might have been answered already somewhere on the Wiki, but if so, I haven't been able to find it.

Initially, I had an assumption that if I create a custom module, download and specify it via the CLI `-d` option, it would be used for _every_ patient generated in that run. However, this does not seem to be the case. For example, I built a simple module where a Device (Implantable defibrillator) was to be implanted in the patient. Only direct transitions used, no conditional or distributed ones. And yet, upon running Synthea to create, say, 10 patients, only 1 out of those 10 contained the expected Device. Why is that? I suppose, my understanding of exactly how those custom modules are evaluated is mistaken. Can you please explain that? (Or if it is in fact already documented somewhere in detail, simply point me there.)

**Comment by jawalonoski** [comment, marked as answer]
My suspicion is that this is related to a set of FAQs where the answer revolves around the `exporter.years_of_history` configuration setting.

I can't say for certain, because I haven't seen your module, but...

If you just have a set of direct transitions, with no delays or guards, your implantable defibrillator is likely being installed on a bunch of newborn babies.

Every module in Synthea runs every single time step, starting with birth of the patient. It is up to the module to determine when and how events occur. If there are no [Delays or Guards](https://github.com/synthetichealth/synthea/wiki/Age%3A-Guard-vs.-Delay), the module will just run straight through the first time it executes until it hits the Terminal state. Once it hits the Terminal state, it will never run for that patient again. So, you'll have a bunch of babies with defibrillators.

Later, those babies will grow up and at the time of exporting their records, the system default only exports the last 10 years of history (with some exceptions, see https://github.com/synthetichealth/synthea/wiki/Common-Configuration#other-export-settings).

**Reply by tamasVarjas** [reply]
Thank you, that was in fact the problem. I changed `years_of_history` to `0` in order to keep the entire history, and indeed, it was included in each patient.
